### PR TITLE
WFCORE-6258 Refactor Model interface from wildfly-clustering-common into wildfly-controller

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFCORE-6258

### CODE REVIEW

**Review by yersan — COMMENTED** [review]
Is there any reason for not making the existing wildfly core subsystems implement this new interface in this PR?

File: controller/src/main/java/org/jboss/as/controller/SubsystemModel.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
```
Comment (yersan) [inline controller/src/main/java/org/jboss/as/controller/SubsystemModel.java:21]: All WildFly Core new files should use Apache license instead of LGPL

**Review by yersan — CHANGES_REQUESTED** [review]

File: controller/src/main/java/org/jboss/as/controller/SubsystemModel.java (hunk 1 above)
Comment (pferraro) [reply controller/src/main/java/org/jboss/as/controller/SubsystemModel.java]: Fixed.

**Comment by pferraro** [comment]
> Is there any reason for not making the existing wildfly core subsystems implement this new interface in this PR?

Enumerating supported model versions for every subsystem is probably best done by the corresponding component owner - as this would otherwise require some time to investigate the history of every subsystem.
That said, I should at least do this for the core components that I own, i.e. discovery.

**Review by yersan — APPROVED** [review]

**Comment by yersan** [comment]
Would you mind adding the Jira number to each commit message?, we don't need to wait for a full CI, since it already passed.

**Comment by pferraro** [comment]
> Would you mind adding the Jira number to each commit message?, we don't need to wait for a full CI, since it already passed.

Done.

**Pull request merged**